rules/properties-order/checkEmptyLineBefore.js

```diff
@@ -22,6 +27,10 @@ module.exports = function checkEmptyLineBefore(firstPropData, secondPropData, sh
 		firstPropSeparatedGroup !== secondPropSeparatedGroup && !secondPropIsUnspecified;
 	const startOfUnspecifiedGroup = !firstPropIsUnspecified && secondPropIsUnspecified;
 
+	// Line threshold logic
+	const belowEmptyLineThreshold =
+		allNodesData.length < sharedInfo.emptyLineMinimumPropertyThreshold;
```

Comment:
I don't, all I really need is the length - but I wasn't quite sure how this information should be provided. It wasn't clear to me where this responsibility should lie. 

At the time I pushed it through so I could solve other problems - I'm really looking for guidance on how this should be handled.

Reply:
In [checkNode](https://github.com/hudochenkov/stylelint-order/blob/9da9f2c7cdd028cf899b29dcfbedbbccadb46929/rules/properties-order/checkNode.js#L71) before `allNodesData.forEach()` add:

```js
const propsCount = node.nodes.filter(item => utils.isProperty(item));
```

Then pass this variable to `checkEmptyLineBefore` instead of `allNodesData`.